### ISSUE TITLE
Training problems

### ISSUE THREAD

**Issue by KYRIEZX** [grammarly/gector#76]
Me again...(╯°Д°)╯ ┻━┻
I set the training parameter n_epoch as some number, but the training always ended after 4 epochs. 
`python train.py --n_epoch 10 --train_set ./train_gec.txt --batch_size 64 --dev_set ./dev2labels.txt --model_dir ./new_model1 --transformer_model xlnet --pretrain_folder xlnetmodel --pretrain xlnet_0_gector --vocab_path data/output_vocabulary --special_tokens_fix 0
`
It ended like this:
![image](https://user-images.githubusercontent.com/56826566/100058986-46d6f800-2e65-11eb-8a21-da7e9cd48a8a.png)

**Comment by KYRIEZX** [comment]
What's more, you mentioned that you use errorful-only sentences in stage 2, so we should discard the sentence pair whose source and target are same when stage 2?

**Comment by skurzhanskyi** [comment]
It looks like your training get stopped because of the early stopping.
In train script we have [patience](https://github.com/grammarly/gector/blob/86a310bee4aa470690f631b80dec832cdecb707a/train.py#L216) parameter. By default, it's 3 what means that it'll stop the training if 3 epochs in the row there are no improvements on the validation set. That's exactly like your situation. You can increase `patience` value, but it looks unhealty.

**Comment by skurzhanskyi** [comment]
> What's more, you mentioned that you use errorful-only sentences in stage 2, so we should discard the sentence pair whose source and target are same when stage 2?

Exactly

**Comment by KYRIEZX** [comment]
> It looks like your training get stopped because of the early stopping.
> In train script we have [patience](https://github.com/grammarly/gector/blob/86a310bee4aa470690f631b80dec832cdecb707a/train.py#L216) parameter. By default, it's 3 what means that it'll stop the training if 3 epochs in the row there are no improvements on the validation set. That's exactly like your situation. You can increase `patience` value, but it looks unhealty.

Is it important to set update_per_epoch == 10000 in stage1? It seems very slow when training on such a big dataset.

**Comment by skurzhanskyi** [comment]
Sure, you can change it.
